### PR TITLE
Set number of pytest processes via env variable

### PR DESCRIPTION
**Pull request by stephendonner**
@oremj / @m8ttyB r?  And, mind holding off on merging, if r+d, until @davehunt or I can confirm we're good-to-go from the Jenkins side?  (Should be soon; thanks!)

Ad-hoc job here: https://qa-preprod-master.fxtest.jenkins.stage.mozaws.net/job/bouncer.adhoc/26/console

### CODE REVIEW

File: tests/e2e/Jenkinsfile [hunk 1]
```diff
@@ -9,7 +9,10 @@ pipeline {
     timeout(time: 5, unit: 'MINUTES')
   }
   environment {
-    PYTEST_ADDOPTS = "-n=10 --color=yes"
+    PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
+    PYTEST_ADDOPTS =
+    "-n=${PYTEST_PROCESSES} " +
+    "--color=yes"
```
Comment (davehunt) [inline tests/e2e/Jenkinsfile]: Could you indent these two lines by two spaces, for consistency.

Comment (stephendonner) [reply tests/e2e/Jenkinsfile]: Yes, sorry!

**Comment by davehunt** [comment]
This can be merged after [bug 1410925](https://bugzilla.mozilla.org/show_bug.cgi?id=1410925) is resolved.

**Comment by m8ttyB** [comment]
Non-blocking, can you explain why this change is necessary?

**Comment by stephendonner** [comment]
Thanks, @m8ttyB - this allows us to dynamically set the concurrency based on the capacity of the host (Jenkins), and will likely be greater than 10, on at least the production CI.

**Pull request merged**